File: gcd/python_gcd/gcd.py
import datetime
import sys
import os
import time

def gcd(a, b):
  if b == 0:
    return a
  else:
    return gcd(b, (a % b))

def main():
  if len(sys.argv) != 1:
    print("Usage: %s <noargs>" % sys.argv[0])
    sys.exit()
  print(gcd(6,4))
  #print(x)
  sys.exit(), end

main()

File: gcd/ruby_gcd/gcd.rb

def gcd(a, b)
  if b == 0 then
    return a
  else
    return gcd(b, (a % b))
  end
end

def main()
  if ARGV.length != 0 then
    puts "Usage: #{$0} <noargs>"
    exit(1)
  end
  puts "got here"
  puts gcd(6,4)
end

main()
